### DISCUSSION TITLE
IDEA: Backup path change button

### DISCUSSION THREAD

**Discussion by johnkennyuk** [keshav-space/safenotes#169] [Ideas]
I recently discovered this gem of an app and have started to use it, after checking the settings i saw and used the backup button but did wonder why it was only backing up to the downloads folder, granted it suggests moving or copying the file elsewhere after but i thought surely it would make so much more sense to be able to have s setting to change the backup path, then i would save it to a cloud storage folder that are normally viewable in android file explorer natively at least the Google drive paths are at a minimum.

So my idea would be to add another setting button under the backup button that allows the backup path to be changed.

**Comment by keshav-space** [comment]
## TL;DR
For **Safe Notes** to store backups at the user's desired location, it would require using `All File Permission`. However, this is not a good idea, especially since, starting from Android 11, Android has shifted towards more privacy-friendly APIs. We will never use `All File Permission` because it is a highly privacy-invasive permission that **Safe Notes** does not need to perform its core functionality.

## Below is a brief summary of how Android's new privacy-friendly API impacts storage access
### Android >= 11

- **No Storage Permission** : Apps can create new files in the download folder without permission
- **Media and File Permission** : Apps can `read` media and files but can't modify or create files except in `app-specific storage space` [ _moreover, the android will never expose the actual file location rather a cached file location is shared with apps_]
- **All File Permission** : Apps can `read`, `modify`, `create`, and `delete` files on the file system

### Android < 11

- **No Storage Permission** :  App can not access any part of external shared storage
- **File Permission** : App can read, modify, create, and delete files on the file system.

## How Safe Notes handles the storage access?
These access controls will become more stringent in the coming year, something similar to iOS, and that's good for user privacy.  
The bottom line is that apps should not use `All file Permission` unless absolutely needed. 
**_Safe Notes does not and will not use `All File Permission`, this means that users will not be able to choose the location of their choice for backup_**. Backup will be made inside the `Downloads` directory within the `Safe Notes` folder.

**Reply by johnkennyuk** [reply]
Can you not use alternative storage apis such as cloud storage providers surely android has secure apis available to directly store files to Google drive or even Microsoft Onedrive? Android file explorers have those options would they still require the permissions you mentioned local storage would require the one you don't want to integrate? I know Google drive should have options to encrypt in transit plus it also differs to Onedrive in that it doesn't have to be synced with local cache, it can effectively work as a mapped drive albeit still has its local cache (going by knowledge of windows Google drive).

Granted I am only making assumptions here as I have no experience with android development, have you investigated cloud storage api options and do they still require that permission type you will not use? I only ask as surely Microsoft or Google have taken such possibilities into account, Microsoft I do know allows setup of some pretty stringent compliance and security policies which include android devices surely if what you say about the permissions you mentioned are that vulnerable they would have settings available to make said vulnerable setups non-compliant?

Thanks for your reply it's appreciated, but cloud storage options would be ideal nowadays.
